FlightReservation/components/signUpButtons.js

```diff
@@ -0,0 +1,140 @@
+    View,
+    Text,
+    Image,
+    Modal,
+    TouchableOpacity,
+} from 'react-native';
+
+function SignUpButton(props) {
+    const { objValues, showModal, setIsLoginFormActive, isLoginFormActive, signUpGoogle } = props;
+    //sign in/log in button styles
+    const [classColor, setClassColor] = useState('#B6B7BA');
+    const [buttonDisable, setButtonDisable] = useState(true);
+    //sign in/log in with google button styles
+    const [classColorGoogle, setClassColorGoogle] = useState('#B6B7BA');
+    const [buttonDisableGoogle, setButtonDisableGoogle] = useState(true);
+
+    React.useEffect(() => {
+        checkNulls();
+        checkNullsGoogle();
+    });
+    const checkNulls = () => {
+        if (
+            !objValues.name ||
+            !objValues.email ||
+            !objValues.password ||
+            !objValues.agreed ||
+            !objValues.subscribed
+        ) {
+            setClassColor('#B6B7BA');
+            setButtonDisable(true);
+        } else {
+            setClassColor('#5B6EF8');
+            setButtonDisable(false);
+        }
+    };
+
+    const checkNullsGoogle = () => {
+        if (!objValues.agreed || !objValues.subscribed) {
+            setClassColorGoogle('#B6B7BA');
+            setButtonDisableGoogle(true);
+        }else{
+            setClassColorGoogle('#5B6EF8');
+            setButtonDisableGoogle(false);
+        }
+    }
+    return (
+        <View style={[styles.containerForm, { marginTop: 20 }]}>
+            <TouchableOpacity
+                disabled={buttonDisable}
+                onPress={showModal}
+                style={[
+                    styles.ContainerOfButtonSignUp,
+                    styles.containerForm,
+                    { backgroundColor: classColor, borderWidth: 0 },
+                ]}>
+                <View>
+                    <Text style={styles.textButtons}>
+                        {isLoginFormActive ? "Log In" : "Sign Up"}
+                    </Text>
+                </View>
+            </TouchableOpacity>
+            <View style={{ alignItems: 'center', margin: 10 }}>
+                <Text style={{ color: '#C9CED6', fontSize: 15 }}>or</Text>
+            </View>
+            <TouchableOpacity disabled={buttonDisableGoogle} onPress={() => signUpGoogle()}>
+                <View
+                    style={[
+                        styles.ContainerOfButtonSignUpGoogle,
+                        { backgroundColor: classColorGoogle },
+                    ]}>
+                    <View style={styles.viewIconGoogle}>
+                        <Image
+                            style={styles.iconGoogle}
+                            source={require('../img/google.png')}
+                        />
+                    </View>
+                    <Text style={[styles.textButtons, { paddingLeft: 25 }]}>
+                        {isLoginFormActive ? "Log In" : "Sign Up"} with Google
+                    </Text>
+                </View>
```

Comment:
User `Prettier` here, there are too much indentation and formating errors here.

Reply:
Done